### PR TITLE
chore(ts): convert sortOptions to typescript

### PR DESCRIPTION
**Pull request by scttcper**
This component had state, but I decided that it didn't deserve to have state.

### CODE REVIEW

File: src/sentry/static/sentry/app/views/issueList/sortOptions.tsx [hunk 1]
```diff
@@ -6,43 +6,28 @@ import DropdownControl, {DropdownItem} from 'app/components/dropdownControl';
 import {t} from 'app/locale';
 import space from 'app/styles/space';
 
-class IssueListSortOptions extends React.PureComponent {
+type Props = {
+  sort?: string;
+  onSelect?: (sort: string) => void;
```
Comment (matejminar) [inline src/sentry/static/sentry/app/views/issueList/sortOptions.tsx:11]: Can we make these required?

File: src/sentry/static/sentry/app/views/issueList/sortOptions.tsx [hunk 2]
```diff
@@ -6,43 +6,28 @@ import DropdownControl, {DropdownItem} from 'app/components/dropdownControl';
...
+};
+
+class IssueListSortOptions extends React.PureComponent<Props> {
```
Comment (matejminar) [inline src/sentry/static/sentry/app/views/issueList/sortOptions.tsx:14]: Should we refactor to function component now that there's no state?

Comment (scttcper) [reply src/sentry/static/sentry/app/views/issueList/sortOptions.tsx]: @matejminar great suggestion

Comment (billyvg) [reply src/sentry/static/sentry/app/views/issueList/sortOptions.tsx]: Iirc function component !== purecomponent

**Review by priscilawebdev — APPROVED** [review]
Looks good :)

File: src/sentry/static/sentry/app/views/issueList/sortOptions.tsx (hunk 2 above)
Comment (matejminar) [reply src/sentry/static/sentry/app/views/issueList/sortOptions.tsx]: Correct, pure component uses shallow comparison on state+props change and prevents unnecessary re-renders.

I probably wouldn't optimize this prematurely though as rerenders in such a simple component are very cheap and overusing PureComponent can sometimes cause more harm than good.

Unless of course, we had some performance issues within this component in the past, in which case we could even move PureComponent up and make `sortOptions`'s parent `IssueListFilters` a PureComponent to encompass all of its children.

What do you think @billyvg ?

Comment (billyvg) [reply src/sentry/static/sentry/app/views/issueList/sortOptions.tsx]: I think everything you said is right! Just wanted to point out that they are not equivalent

**Review by matejminar — APPROVED** [review]

**Pull request merged**